plugin/evm/tx_gossip_test.go

```diff
@@ -0,0 +1,254 @@
+	require.NoError(blk.Accept(context.Background()))
+	<-importAccepted
+
+	// sender for the peer requesting gossip from [vm]
+	ctrl := gomock.NewController(t)
+	peerSender := common.NewMockSender(ctrl)
+	router := p2p.NewRouter(logging.NoLog{}, peerSender, prometheus.NewRegistry(), "")
+
+	// we're only making client requests, so we don't need a server handler
+	client, err := router.RegisterAppProtocol(ethTxGossipProtocol, nil, nil)
+	require.NoError(err)
+
+	emptyBloomFilter, err := gossip.NewBloomFilter(txGossipBloomMaxItems, txGossipBloomFalsePositiveRate)
+	require.NoError(err)
+	emptyBloomFilterBytes, err := emptyBloomFilter.Bloom.MarshalBinary()
+	require.NoError(err)
+	request := &sdk.PullGossipRequest{
+		Filter: emptyBloomFilterBytes,
+		Salt:   utils.RandomBytes(32),
+	}
+
+	requestBytes, err := proto.Marshal(request)
+	require.NoError(err)
+
+	wg := &sync.WaitGroup{}
+
+	requestingNodeID := ids.GenerateTestNodeID()
+	peerSender.EXPECT().SendAppRequest(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Do(func(ctx context.Context, nodeIDs set.Set[ids.NodeID], requestID uint32, appRequestBytes []byte) {
+		go func() {
+			require.NoError(vm.AppRequest(ctx, requestingNodeID, requestID, time.Time{}, appRequestBytes))
+		}()
+	}).AnyTimes()
+
+	sender.SendAppResponseF = func(ctx context.Context, nodeID ids.NodeID, requestID uint32, appResponseBytes []byte) error {
+		go func() {
+			require.NoError(router.AppResponse(ctx, nodeID, requestID, appResponseBytes))
+		}()
+		return nil
+	}
+
+	// we only accept gossip requests from validators
+	mockValidatorSet, ok := vm.ctx.ValidatorState.(*validators.TestState)
+	require.True(ok)
+	mockValidatorSet.GetCurrentHeightF = func(context.Context) (uint64, error) {
+		return 0, nil
+	}
+	mockValidatorSet.GetValidatorSetF = func(context.Context, uint64, ids.ID) (map[ids.NodeID]*validators.GetValidatorOutput, error) {
+		return map[ids.NodeID]*validators.GetValidatorOutput{requestingNodeID: nil}, nil
+	}
+
+	// Ask the VM for any new transactions. We should get nothing at first.
+	wg.Add(1)
+	onResponse := func(_ context.Context, nodeID ids.NodeID, responseBytes []byte, err error) {
+		require.NoError(err)
+
+		response := &sdk.PullGossipResponse{}
+		require.NoError(proto.Unmarshal(responseBytes, response))
+		require.Empty(response.Gossip)
+		wg.Done()
+	}
+	require.NoError(client.AppRequest(context.Background(), set.Set[ids.NodeID]{vm.ctx.NodeID: struct{}{}}, requestBytes, onResponse))
+	wg.Wait()
+
+	// Issue a tx to the VM
+	address := testEthAddrs[0]
+	key := testKeys[0].ToECDSA()
+	tx := types.NewTransaction(0, address, big.NewInt(10), 100_000, big.NewInt(params.LaunchMinGasPrice), nil)
+	signedTx, err := types.SignTx(tx, types.NewEIP155Signer(vm.chainID), key)
+	require.NoError(err)
+
+	errs := vm.txPool.AddLocals([]*types.Transaction{signedTx})
+	require.Len(errs, 1)
+	require.Nil(errs[0])
+
+	// wait so we aren't throttled by the vm
+	time.Sleep(throttlingPeriod / throttlingLimit)
```

Comment:
Can we replace this in the future with a direct function call that tells the throttler X amount of time has passed?

Reply:
Updated it in the latest commit